modules/rostests/apitests/shell32/InternalExtractIconListW.cpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * PROJECT:     ReactOS API tests
+ * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:     Tests for InternalExtractIconListW
+ * COPYRIGHT:   Copyright 2025 Katayama Hirofumi MZ (katayama.hirofumi.mz@gmail.com)
+ */
+
+#include "shelltest.h"
+#include <undocshell.h>
+#include <versionhelpers.h>
+
+START_TEST(InternalExtractIconListW)
+{
+    if (IsWindowsVistaOrGreater())
+    {
+        skip("InternalExtractIconListW of Vista+ is useless (returns NULL)\n");
+        return;
+    }
+
+    WCHAR szPath[MAX_PATH];
+    GetModuleFileNameW(NULL, szPath, _countof(szPath));
+
+    HGLOBAL hPairs = InternalExtractIconListW(GetModuleHandleW(NULL), szPath, 0);
+    ok(hPairs != NULL, "hPairs was NULL\n");
+
+    UINT nIcons = (UINT)(GlobalSize(hPairs) / sizeof(ICON_AND_ID));
+    ok(nIcons != 0, "nIcons was zero\n");
+
+    PICON_AND_ID pPairs = (PICON_AND_ID)GlobalLock(hPairs);
+    ok(pPairs != NULL, "pPairs was NULL\n");
+
+    ok((pPairs && pPairs[0].hIcon != NULL), "pPairs[0].hIcon was NULL\n");
+    ok((pPairs && pPairs[0].nIconID != 0), "pPairs[0].nIconID was zero\n");
+
+    if (pPairs)
+    {
```

Comment:
```suggestion
    if (pPairs)
    {
        ok((pPairs[0].hIcon != NULL), "pPairs[0].hIcon was NULL\n");
        ok((pPairs[0].nIconID != 0), "pPairs[0].nIconID was zero\n");

```

Reply:
The counter of failures will become wrong. Rejected.